### PR TITLE
DMSO - Chem that causes injection on touch with chems

### PR DESCRIPTION
**Pull request by Azrun**
[feature][input wanted]
## About the PR 
Adds DMSO, a miscible liquid that causes chemicals to be readily absorbed through the skin.  For better or for worse!

Adds Dimethyl sulfide:
* Smells bad
* Precursor chem

Adds Dimethyl sulfoxide:
* When splashed, all other reagents that are part of the splash have a chance of penetrating the targets skin. (Factors in clothing)
  * "Splash/Touch" effect has 5 second cooldown.  
  * Units transferred limited to minimum of DMSO amount splashed (to scale on DMSO mixture), volume of reagent splashed (based on other reagent amount), and 20 (hard cap because water tanks).
  * Effectiveness of transfer based on gear jobs spawn vs chem type (hypospray list vs everything else). 
image

* Mild analgesic 

Fluid Adjustments:
* Fluids on turf with units 2 or less will be ignored by shoes.
* Pre-calculate fluid depth on creation and addition of fluid/fluid-groups to ensure entering a fluid determines appropriate clothing is checked for chem reactions.

## Why's this needed? 
Provides a different vector for injection.  Intended to give doctors an alternative means of administrating drugs but also offers a breadth of "opportunities" to antags.

```changelog
(u)Azrun:
(*)Adds DMSO, a chemical that allows splashed reagents to be absorbed through your skin.... 
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-Medical.dm [hunk 1]
```diff
@@ -1557,3 +1557,41 @@ datum
 					L.contract_disease(/datum/ailment/disease/food_poisoning, null, null, 1)
 				..()
 				return
+
+		medical/dimethyl_sulfoxide
+			name = "DMSO"
```
Comment (BenLubar) [inline code/modules/chemistry/Reagents-Medical.dm:1562]: I'd use the full name, like we do with `potassium iodide`.

Comment (Azrun) [reply code/modules/chemistry/Reagents-Medical.dm:1562]: Hmm... potassium iodide ~= iodine. Colloquially it is common, but from a chemistry context it seems misleading or incorrect. 

In the spirit of getting people to learn things through SS13, I'm down with this suggestion.

**Comment by Azrun** [comment]
If this feature end up being a no, I can also make a separate PR for the fluids changes if there is interest.

**Comment by ZeWaka** [comment]
@Azrun fluid change PR would be lovely

**Pull request closed without merging**